metadata-ingestion/src/datahub/ingestion/source/clickhouse.py

```diff
@@ -0,0 +1,301 @@
+    ByteFixedString,
+)
+from clickhouse_driver.columns.tuplecolumn import TupleColumn
+from clickhouse_driver.columns.uuidcolumn import UUIDColumn
+from clickhouse_driver.errors import UnknownTypeError
+
+from datahub.configuration.common import (
+    ConfigModel,
+    AllowDenyPattern,
+)
+from datahub.emitter.mce_builder import (
+    DEFAULT_ENV,
+    make_data_platform_urn,
+    make_dataset_urn,
+)
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.api.source import (
+    Source,
+    SourceReport,
+)
+from datahub.ingestion.source.metadata_common import MetadataWorkUnit
+from datahub.metadata.com.linkedin.pegasus2avro.common import AuditStamp
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import DatasetSnapshot
+from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    SchemaField,
+    SchemaMetadata,
+    SchemaFieldDataType,
+)
+from datahub.metadata.schema_classes import (
+    ArrayTypeClass,
+    DateTypeClass,
+    TimeTypeClass,
+    NumberTypeClass,
+    EnumTypeClass,
+    RecordTypeClass,
+    MapTypeClass,
+    StringTypeClass,
+    OtherSchemaClass,
+    NullTypeClass,
+)
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ClickhouseConfig(ConfigModel):
+    host: str = "localhost"
+    username: Optional[str] = None
+    password: Optional[str] = None
+    env: str = DEFAULT_ENV
+    client_kwargs: dict = {}
+    schema_pattern: AllowDenyPattern = AllowDenyPattern.allow_all()
+    table_pattern: AllowDenyPattern = AllowDenyPattern.allow_all()
+
+
+class ClickhouseSource(Source):
+    config: ClickhouseConfig
+    report: SourceReport
+
+    def __hash__(self):
+        return id(self)
+
+    def __init__(self,
+                 ctx: PipelineContext,
+                 config: ClickhouseConfig):
+        super().__init__(ctx)
+        self.config = config
+        self.report = SourceReport()
+
+        self._client = client = Client(host=self.config.host,
+                                       user=self.config.username,
+                                       password=self.config.password,
+                                       **self.config.client_kwargs)
+
+    @classmethod
+    def create(cls,
+               config_dict: dict,
+               ctx: PipelineContext) -> Source:
+        config = ClickhouseConfig.parse_obj(config_dict)
+        return cls(ctx,
+                   config)
+
+    def get_workunits(self) -> Iterable[MetadataWorkUnit]:
+        platform = "clickhouse"
+        databases = self._client.execute('show databases')
+        for database, in databases:
+            if not self.config.schema_pattern.allowed(database):
+                logger.info(f'Skipping database: {database}')
+                continue
+
+            tables = self._client.execute(f'show tables from {database}')
+            for table, in tables:
+                print(table)
```

Comment:
Nit: maybe convert this to a `.info()` and add some description so we know it's a table being printed.